02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,371 @@
+head(df$Gene)
+```
+
+Enembl gene identifiers have different species-specific prefixes. 
+In zebrafish, Ensembl gene identifiers begin with `ENSDARG` (in human, `ENSG`, etc.).
+
+Now let's do the conversion!
+We'll use the `zebrafish` data from `hcop`; this is what `zebrafish` looks like.
+
+```{r}
+zebrafish
+```
+
+We're interested in the `human_symbol` and `zebrafish_ensembl` columns specifically.
+
+```{r}
+human_zebrafish_key <- zebrafish %>%
+  # we'll want to subset zebrafish to only the columns we're interested in
+  dplyr::select(zebrafish_ensembl, human_symbol)
+
+# Now let's join the expression data from `df` for the subset of genes in `df`
+human_zebrafish_mapped_df <- df %>%
+  dplyr::left_join(human_zebrafish_key, by = c("Gene" = "zebrafish_ensembl"))
+```
+
+Here's what the new data frame looks like:
+
+```{r}
+head(human_zebrafish_mapped_df, n = 25)
+```
+
+Let's get a summary of the gene symbols returned in the `human_symbol` column of our mapped data frame.
+
+```{r}
+# We can use the `count()` function to get a count of the distribution of values in the `human_symbol` column
+human_zebrafish_mapped_df %>%
+  dplyr::group_by(human_symbol) %>%
+  dplyr::count() %>%
+  # Sort by highest `n` which would be the human gene symbol with the most mapped zebrafish Ensembl IDs
+  dplyr::arrange(desc(n))
+```
+
+Looks like we have mapped symbols!
+We also have 415 NAs, which is okay, because we do not expect everything to map across species.
+
+## Sort out multi mappings
+
+* Multiple zebrafish Ensembl gene IDs map to the same human symbol which results in duplicated human gene symbols.
+Some downstream tools may require the data be simplified to 1:1 mappings.
+Using another gene identifier may result in a better 1:1 ratio to the zebrafish Ensembl IDs, i.e. human Ensembl IDs instead of human gene symbols.
+See our [gene ID annotation microarray notebook](https://alexslemonade.github.io/refinebio-examples/02-microarrary/gene-id-annotation_microarray_01_ensembl.html) for an example of mapping between gene identifiers.
+
+* If a zebrafish Ensembl gene ID maps to multiple human symbols, the gene expression values get duplicated.
+Let's look at the `ENSDARG00000069142` example below.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  dplyr::filter(Gene == "ENSDARG00000069142") %>%
+  dplyr::select(Gene, human_symbol, dplyr::everything())
+```
+
+* On the other, if a human gene symbol maps to multiple zebrafish Ensembl IDs, the gene expression values do not necessarily get duplicated.
+Let's look at the `A1BG` example below.
+
+```{r}
+human_zebrafish_mapped_df %>%
+  dplyr::filter(human_symbol == "MATR3") %>%
+  dplyr::select(Gene, human_symbol, dplyr::everything())
+```
+
+## Collapse multiple mapped human gene symbols
+
+We can collapse the multi-mapped values into a list for each Ensembl ID or we can keep it in the format above with one row for each mapped ID (resulting in duplicated rows for some Ensembl IDs).
```

Comment:
Since these multi-mappings can go either way `1zebra:manyhumans` or `manyzebras:1human`, we should try to make this wording more clear. 

Using a consistent wording/notation would be good. I think what I used above is kinda silly, but something like it might be good.